src/Version/Version.cpp

```diff
@@ -82,14 +82,23 @@ std::string getOnnxMlirFullVersion(bool toIncludeLLVM) {
   llvm::raw_string_ostream os(buf);
 #ifdef ONNX_MLIR_VENDOR
   os << ONNX_MLIR_VENDOR;
+#else
+  os << "onnx-mlir product version";
+#endif
+#ifdef ONNX_MLIR_PRODUCT_VERSION
+  os << " " << ONNX_MLIR_PRODUCT_VERSION << '\n';
+#else
+  os << " " << PRODUCT_VERSION_MAJOR << '.';
+  os << PRODUCT_VERSION_MINOR << '.';
+  os << PRODUCT_VERSION_PATCH << '-' << PRODUCT_ID << '\n';
```

Comment:
Aren't the variables in this ELSE derived from `ONNX_MLIR_PRODUCT_VERSION`? I'm not sure if it's expected for them to be set independently of that. So what I think would happen is if `ONNX_MLIR_PRODUCT_VERSION` is unset (which it will be for community builds) then you'll always come down this ELSE and print `0.0.0-NOT SPECIFIED` (if I'm reading the flow correctly in https://github.com/onnx/onnx-mlir/blob/main/src/Compiler/CMakeLists.txt)

Reply:
I think I resolved everything from our last conversation.... let me know your thoughts now.

Reply:
This is what it looks like when the vendor and product number is specified:

```
root@d7f7476a247e:~/onnx-mlir/build# onnx-mlir --version
IBM Z Deep Learning Compiler 4.0.0-zdlc
onnx-mlir version 0.3.1, onnx version 1.11.0 (https://github.com/hamptonm1/onnx-mlir.git 40cdc98e1f4062fea9cf84e6a57c01dbef93f4c2)
LLVM version 16.0.0git (https://github.com/llvm/llvm-project.git 59548fe873d8d98e359fb21fbb2a0852fed17ff5)
DEBUG build with assertions.
Default target: x86_64-unknown-linux-gnu
Host CPU: skylake
```